src/medCore/gui/toolboxes/medToolBox.cpp

```diff
@@ -308,3 +308,30 @@ void medToolBox::toXMLNode(QDomDocument* doc, QDomElement* currentNode)
 	currentNode->appendChild(elmt);
 }
 
+void medToolBox::handleDisplayError(int error)
+{
+    switch (error)
+    {
+    case medAbstractProcess::PIXEL_TYPE:   //! Handle volume errors: pixel type
+        displayMessageError("Pixel type not yet implemented");
+        break;
+    case medAbstractProcess::DIMENSION_3D: //! Handle volume errors: dimension
+        displayMessageError("This toolbox is designed to be used with 3D volumes");
+        break;
+    case medAbstractProcess::DIMENSION_4D: //! Handle volume errors: dimension
+        displayMessageError("This toolbox is designed to be used with 4D volumes");
+        break;
+    case medAbstractProcess::MESH_TYPE:    //! Handle mesh errors: data type
+        displayMessageError("This toolbox is designed to be used with meshes");
```

Comment:
I thought about that point indeed. I don't know, and it's the same idea with the other sentences. If your tlbx needs 1 mesh, "This toolbox requires (at least) one mesh" would be complicated to understand. "with meshes" is more generic i would say. And since it's generic (we do not say "to be used with 2 meshes" for instance), it reminds the user to check his/her data. 

However, if you want a specific message, either we can add a new one here, or you can use `displayMessageError("blablabla");` in your code :)